datalad/plugin/addurls.py

```diff
@@ -218,6 +219,108 @@ def filter_legal_metafield(fields):
     return legal
 
 
+class AnnexKeyParser(object):
```

Comment:
I'm aware of `get_size_from_key`, but I don't think it's an improvement to introduce a more general parser that would cover both use cases.  By nature of being a general parser, it would have to be extended to cover use cases we don't yet have (or sit in AnnexRepo as a half-baked parser, which I don't think is a good thing).

Reply:
> By nature of being a general parser, it would have to be extended to cover use cases we don't yet have

not really -- could provide (and document) parsing of only size and checksum.  IMHO it would still be better than breeding "specialized" parsers around the code base (I believe I saw some additional ones somewhere else, didn't look too wide across extensions though) for every "use case".  Well -- I guess we will just agree to disagree for now.